### PR TITLE
feat(docs): Adding a guide for cloudflare

### PR DESCRIPTION
**Pull request by rmarscher**
Here is a guide with my current understanding of how to run develop and deploy Waku for Cloudflare Pages in the next release that includes the unstable API to access the Hono context.

### CODE REVIEW

File: docs/guides/cloudflare.mdx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+After building, you can test your build by running `npx wrangler pages dev dist` or deploy it to Cloudflare using `npx wrangler pages dev deploy`.
+
+## Notes on Cloudflare's workerd Runtime
+
+Cloudflare does not run NodeJS on their servers. Instead, they use their custom JavaScript runtime called [workerd](https://github.com/cloudflare/workerd).
+
+By default, workerd does not support built-in NodeJS APIs, but support can be added by editing the `compatibility_flags` in your `wrangler.toml` file. Cloudflare does not support all APIs, but the list is growing. For more information, see https://developers.cloudflare.com/workers/runtime-apis/nodejs/ and https://developers.cloudflare.com/workers/configuration/compatibility-dates/#setting-compatibility-flags.
+
+Waku attempts to stay minimal and compatible with [WinterGC servers](https://wintercg.org/). The Node stream API is currently used by Waku, so only the `nodejs_als` compatibility flag is added. If you experience errors in server-side dependencies due to missing NodeJS APIs, try changing this flag to `nodejs_compat` and rebuilding your project.
```
Comment (dai-shi) [inline docs/guides/cloudflare.mdx:25]: We don't use Node Stream API.

```suggestion
Waku attempts to stay minimal and compatible with [WinterGC servers](https://wintercg.org/). The Node AsyncLocalStorage API is currently used by Waku, so only the `nodejs_als` compatibility flag is added. If you experience errors in server-side dependencies due to missing NodeJS APIs, try changing this flag to `nodejs_compat` and rebuilding your project.
```

**Review by dai-shi — COMMENTED** [review]
Thanks for writing the guide!

File: docs/guides/cloudflare.mdx (hunk 1 above)
Comment (rmarscher) [reply docs/guides/cloudflare.mdx]: Thank you for correcting that!

File: docs/guides/cloudflare.mdx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## Accessing Cloudflare Bindings, Execution Context, and Request/Response Objects
+
+Waku currently uses [Hono](https://hono.dev/) in the [advanced mode worker function](https://developers.cloudflare.com/pages/functions/advanced-mode/) it creates for Cloudflare Pages when you build your project.
+
+You can access Cloudflare resources through the [Hono context](https://hono.dev/docs/api/context):
+
+```ts
+import type { Context } from 'hono';
+import { getHonoContext } from 'waku/unstable_hono';
```
[the remark this answers is not in the document]
Comment (rmarscher) [reply docs/guides/cloudflare.mdx]: The guide has been updated to use hono context storage.

**Comment by rmarscher** [comment]
> can you run prettier?

I think I fixed it. Sorry, I have been using biome for other projects and it keeps getting in the way. Should I move this from `docs/guides` to `docs/builder`?

**Comment by dai-shi** [comment]
Still failing?

I think guides is fine, as it's not only about "builder" or "deploy". Maybe we should mv builder/* -> guides/* ?

**Comment by rmarscher** [comment]
🙄 I missed a line in the other code block. I was trying to copy the code block out, format it and copy it back in because vscode wasn't formatting it within the markdown... turns out I think it's because I installed an MDX extension.

Anyway... `pnpx prettier -c --write docs/guides/cloudflare.mdx` is way easier. Hopefully this is all set now.

**Review by dai-shi — APPROVED** [review]

**Pull request merged**